src/internal/ssm.ts

```diff
@@ -0,0 +1,192 @@
+import { JSONObject } from 'k6'
+import http, { RefinedResponse, ResponseType } from 'k6/http'
+
+import { AWSClient, AWSRequest } from './client'
+import { AWSError } from './error'
+import { AWSConfig } from './config'
+import { InvalidSignatureError, URIEncodingConfig } from './signature'
+import { HTTPMethod, HTTPHeaders } from './http'
+
+/**
+ * Class allowing to interact with Amazon AWS's Systems Manager service
+ */
+export class SystemsManagerClient extends AWSClient {
+    method: HTTPMethod
+    commonHeaders: HTTPHeaders
+
+    /**
+     * Create a SystemsManagerClient
+     * @param  {AWSConfig} awsConfig - configuration attributes to use when interacting with AWS' APIs
+     */
+    constructor(awsConfig: AWSConfig) {
+        const URIencodingConfig = new URIEncodingConfig(true, false)
+        super(awsConfig, 'ssm', URIencodingConfig)
+
+        // All interactions with the Systems Manager service
+        // are made via the POST method.
+        this.method = 'POST'
+        this.commonHeaders = {
+            'Accept-Encoding': 'identity',
+            'Content-Type': 'application/x-amz-json-1.1',
+        }
+    }
+
+    /**
+     * Retrieves a parameter from Amazon Systems Manager
+     *
+     * @param {string} name - The ARN or name of the parameter to retrieve.
+     * @param {boolean} withDecryption - The ARN or name of the parameter to retrieve.
+     * @returns {Parameter} - returns the content of the fetched Parameter object.
+     * @throws {SystemsManagerServiceError}
+     * @throws {InvalidSignatureError}
+     */
+    getParameter(name: string, withDecryption: boolean = false): Parameter | undefined {
+        const body = JSON.stringify({ Name: name, WithDecryption: withDecryption })
+
+        // Ensure to include the desired 'Action' in the X-Amz-Target
+        // header field, as documented by the AWS API docs.
+        const signedRequest: AWSRequest = super.buildRequest(
+            this.method,
+            this.host,
+            '/',
+            '',
+            body,
+            {
+                ...this.commonHeaders,
+                'X-Amz-Target': `AmazonSSM.GetParameter`,
+            }
+        )
+
+        const res = http.request(this.method, signedRequest.url, body, {
+            headers: signedRequest.headers,
+        })
+        this._handle_error('GetParameter', res)
+        // a little confusing but need to unnest Parameter...
+        // TODO: make Parameter the root type with ParameterItem as a subtype
```

Comment:
this change is now implemented and it is technically more accurate to what AWS responds with, but makes it a little cumbersome as now we have to unwrap what we care about from a parameter object. i'm thinking stick with it for now. it can be changed if it proves to be a burden.